lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingEvaluation.cxx

```diff
@@ -201,15 +247,56 @@ struct KrigingEvaluationSampleFunctor
   } // operator()
 }; // struct KrigingEvaluationSampleFunctor
 
+struct KrigingEvaluationSampleFunctor1D
+{
+  const Sample & input_;
+  Sample & output_;
+  const KrigingEvaluation & evaluation_;
+  UnsignedInteger trainingSize_;
+
+  KrigingEvaluationSampleFunctor1D(const Sample & input,
+                                   Sample & output,
+                                   const KrigingEvaluation & evaluation)
+    : input_(input)
+    , output_(output)
+    , evaluation_(evaluation)
+    , trainingSize_(evaluation.inputSample_.getSize())
+  {
+    // Nothing to do
+  }
+
+  inline void operator()( const TBB::BlockedRange<UnsignedInteger> & r ) const
+  {
+    const UnsignedInteger inputDimension = input_.getDimension();
+    const UnsignedInteger start = r.begin();
+    const UnsignedInteger size = r.end() - start;
+    Scalar value = 0.0;
+    for (UnsignedInteger i = 0; i != size; ++i)
+    {
+      for (UnsignedInteger j = 0; j < trainingSize_; ++j)
+        value += evaluation_.covarianceModel_.getImplementation()->computeAsScalar(input_.getImplementation()->data_begin() + (start + i) * inputDimension, evaluation_.inputSample_.getImplementation()->data_begin() + (j * inputDimension) ) * evaluation_.gamma_(j, 0);
+      output_(start + i, 0) = value;
+    }
+  } // operator()
```

Comment:
this loop would be simpler to iterate on the global indices (begin, end), instead of (0, size) as we never need the local index : we always use start + i !

Reply:
correct. I am wondering if it would be better to increment a counter by inputDimension instead of doing a multiplication at each step.

Reply:
I'd say dont bother: the improvement should be too tiny compared to the floating point operations that cost much more. Also the compiler should already optimize this kind of stuff.